### PR TITLE
Test for action description of code actions, and simplify description for extracting method to file

### LINKED ISSUE #17998: Extract method refactoring should not show full file path
**TypeScript Version:**  2.5.1

**Code**

```ts
function foo() {
    return 1 + 2;
}
```

highlight `1 + 2` and trigger code actions

**Bug:**
For the file extract, we show the entire file path:

screen shot 2017-08-23 at 5 39 45 pm

This path can be very long and is not necessary since extract can currently only operate within the same file (at least to my knowledge).

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #17998

### CODE REVIEW

File: src/harness/fourslash.ts [hunk 1]
```diff
@@ -2761,20 +2761,25 @@ namespace FourSlash {
...
+                }
             }
-            else if (!negative && !isAvailable) {
-                this.raiseError(`verifyApplicableRefactorAvailableForRange failed - expected a refactor but found none.`);
+            else {
+                if (!isAvailable) {
+                    this.raiseError(`verifyApplicableRefactorAvailableForRange failed - expected a refactor but found none.`);
+                }
+                if (refactors.length > 1) {
+                    this.raiseError(`2 available refactors both have name ${name} and action ${actionName}`);
```
Comment (amcasey) [inline src/harness/fourslash.ts]: Consider using the actual number.

File: src/services/refactors/extractMethod.ts [hunk 2]
```diff
@@ -560,32 +560,32 @@ namespace ts.refactor.extractMethod {
...
-                ? `class ${scope.name.text}`
+                ? `class '${scope.name.text}'`
                 : scope.name.text
-                    ? `class expression ${scope.name.text}`
+                    ? `class expression '${scope.name.text}'`
                     : "anonymous class expression";
         }
         else if (isSourceFile(scope)) {
-            return `file '${scope.fileName}'`;
+            return "this file";
```
Comment (amcasey) [inline src/services/refactors/extractMethod.ts]: What about "file scope" or "top-level scope"?

**Comment by amcasey** [comment]
FYI @RyanCavanaugh

File: src/services/refactors/extractMethod.ts (hunk 2 above)
Comment (andy-ms) [inline src/services/refactors/extractMethod.ts]: Maybe this should be "global scope" for a global and "module scope" for a module? @DanielRosenwasser thoughts?

**Pull request merged**